x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LocalPhysicalPlanOptimizer.java

```diff
@@ -230,7 +230,7 @@ public static boolean canPushToSource(Expression exp) {
 
         private static boolean isAttributePushable(Expression expression, ScalarFunction operation) {
             if (expression instanceof FieldAttribute f && f.getExactInfo().hasExact()) {
-                return true;
+                return f.exactAttribute().field().isAggregatable();
```

Comment:
Extract a simple (static) method that works on expression to reuse it here and in canPushDownOrders.